alps-ios-app-color/ViewController.swift

```diff
@@ -9,46 +9,63 @@
 import UIKit
 import AlpsSDK
 
-class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource{
+class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, MatchDelegate {
     
     @IBOutlet weak var rangeSlider: UISlider!
     @IBOutlet weak var rangeLabel: UILabel!
     @IBOutlet weak var colorPicker: UIPickerView!
     @IBOutlet weak var publishButton: UIButton!
-    var pickerData : [String] = [String]()
-    var currentPub : Publication!
+    var pickerData: [String] = [String]()
+    var onMatch: OnMatchClosure?
+    
+    let colors = ["yellow": UIColor.yellow,
+                  "red": UIColor(red: 255/255, green: 106/255, blue: 93/255, alpha: 1),
+                  "orange": UIColor(red: 255/255, green: 172/255, blue: 62/255, alpha: 1),
+                  "magenta": UIColor(red: 255/255, green: 116/255, blue: 217/255, alpha: 1),
+                  "lightgray": UIColor.lightGray,
+                  "green": UIColor(red: 134/255, green: 255/255, blue: 150/255, alpha: 1),
+                  "cyan": UIColor.cyan,
+                  "teal": UIColor(red: 20/255, green: 229/255, blue: 204/255, alpha: 1),
+                  "pink": UIColor(red: 255/255, green: 173/255, blue: 171/255, alpha: 1),
+                  "white": UIColor(red: 255/255, green: 255/255, blue: 255/255, alpha: 1)
+    ]
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        // Start Monitoring Matches
+        self.onMatch = { matches, _ in
+            print("🏔 You've got new matches!!! 🏔\n\(matches.map { $0.publication?.encodeToJSON() })")
+            // Color the background according to the match
+            let mostRecentDate = matches.max(by: {
+                $0.createdAt! < $1.createdAt!
+            })
+            guard let properties = mostRecentDate?.publication?.properties else {
+                print("No properties.")
+                return
+            }
+            let color = properties["color"] as? String
```

Comment:
you should also guard let color. Line below you have to force unwrap it with `!` which is not good practice.